samples/snippets/quickstart.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A sample script showing how to create, list and delete Google Compute Engine
+instances using the google-cloud-compute library. It can be run from command
+line to create, list and delete an instance in a given project in a given zone.
+"""
+
+import argparse
+
+# [START compute_instances_list]
+# [START compute_instances_list_all]
+# [START compute_instances_create]
+# [START compute_instances_delete]
+# [START compute_instances_operation_check]
+import typing
+
+import google.cloud.compute_v1 as gce
+
+# [END compute_instances_operation_check]
+# [END compute_instances_delete]
+# [END compute_instances_create]
+# [END compute_instances_list_all]
+# [END compute_instances_list]
+
+
+# [START compute_instances_list]
+def list_instances(project: str, zone: str) -> typing.Iterable[gce.Instance]:
+    """
+    Gets a list of instances created in given project in given zone.
+    Returns an iterable collection of Instance objects.
+
+    Args:
+        project: Name of the project you want to use.
+        zone: Name of the zone you want to check, for example: us-west3-b
+
+    Returns:
+        An iterable collection of Instance objects.
+    """
+    instance_client = gce.InstancesClient()
+    instance_list = instance_client.list(project=project, zone=zone)
+    return instance_list
+
+
+# [END compute_instances_list]
+
+
+# [START compute_instances_list_all]
+def list_all_instances(project: str) -> typing.Dict[str, typing.Iterable[gce.Instance]]:
+    """
+    Returns a dictionary of all instances present in a project, grouped by their zone.
+
+    Args:
+        project: Name of the project you want to use.
+
+    Returns:
+        A dictionary with zone names as keys (in form of "zones/{zone_name}") and
+        iterable collections of Instance objects as values.
+    """
+    instance_client = gce.InstancesClient()
+    agg_list = instance_client.aggregated_list(project=project)
+    all_instances = {}
+    for zone, response in agg_list:
+        if response.instances:
+            all_instances[zone] = response.instances
+    return all_instances
+
+
+# [END compute_instances_list_all]
+
+
+# [START compute_instances_create]
+def create_instance(
+    project: str, zone: str, machine_type: str, machine_name: str, source_image: str
```

Comment:
@m-strzelczyk please use default values in params and naming convention as used in PHP: https://github.com/GoogleCloudPlatform/php-docs-samples/blob/80801c983457f407cb9a81d3b60654369f090e03/compute/cloud-client/instances/src/create_instance.php#L47

This is to make sure we have alignment between different langagues.